### PR TITLE
#166491401 Fix calendar invite flow

### PR DESCRIPTION
**Pull request by andrewinsoul**
#### What Does This PR Do?
This PR fixes the calendar invite flow when user attends an event from the application

#### Description Of Task To Be Completed
- Fix duplication of event when the user attends an event from the application
- Mark the event as accepted when the user accepts the event through the application
- Add tests

#### Any Background Context You Want To Provide?
None

#### How can this be manually tested?
- Fetch the branch locally by running `git fetch origin bg-fix-calendar-invite-flow-166491401` & `git checkout bg-fix-calendar-invite-flow-166491401` respectively
- Create an event in the application
- Let another user login to the application and attend the event created earlier in the application. The event should be added into the attendee's calendar and also updated to attending

#### What are the relevant pivotal tracker stories?
[#166491401](https://www.pivotaltracker.com/story/show/166491401)

### CODE REVIEW

File: server/graphql_schemas/tests/events/snapshots/snap_test_mutations.py [hunk 1]
```diff
@@ -166,26 +166,42 @@
...
     'data': {
-        'updateEvent': {
-            'actionMessage': 'Event Update is successful.',
-            'updatedEvent': {
-                'id': 'RXZlbnROb2RlOjU=',
-                'title': "This is a test don't panic."
-            }
+        'updateEvent': None
+    },
+    'errors': [
```
Comment (Alameen688) [inline server/graphql_schemas/tests/events/snapshots/snap_test_mutations.py:171]: Snapshot contains errors. Kindly help look into that

File: server/graphql_schemas/tests/events/snapshots/snap_test_mutations.py [hunk 2]
```diff
@@ -166,26 +166,42 @@
...
     'data': {
-        'updateEvent': {
-            'actionMessage': 'Event Update is successful.',
-            'updatedEvent': {
-                'id': 'RXZlbnROb2RlOjU=',
-                'title': 'Not really a party'
-            }
+        'updateEvent': None
+    },
+    'errors': [
```
Comment (Alameen688) [inline server/graphql_schemas/tests/events/snapshots/snap_test_mutations.py:191]: Snapshot contains errors. Kindly help look into that.

File: server/graphql_schemas/tests/interest/snapshots/snap_test_mutations.py [hunk 3]
```diff
@@ -75,36 +75,49 @@
...
-        'unjoinCategory': {
-            'unjoinedCategory': {
-                'followerCategory': {
-                    'description': 'For people who want to be happy.',
-                    'id': 'Q2F0ZWdvcnlOb2RlOjI=',
-                    'name': 'Python Meetup'
-                },
-                'id': 'SW50ZXJlc3ROb2RlOk5vbmU='
-            }
+    'errors': [
```
Comment (Alameen688) [inline server/graphql_schemas/tests/interest/snapshots/snap_test_mutations.py:78]: Snapshot contains errors. Kindly help look into that.

**Review by Alameen688 — COMMENTED** [review]
Great job @andrewinsoul! Kindly help address the comments.

File: server/graphql_schemas/utils/helpers.py [hunk 4]
```diff
@@ -213,5 +213,24 @@ def add_event_to_calendar(andela_user, event):
...
     event_details = build_event(event, attendees)
-    return calendar.events().insert(
+    created_event = calendar.events().insert(
             calendarId='primary', body=event_details).execute()
+    event.event_id_in_calendar = created_event['id']
+    event.save()
+
+
+def update_event_status_on_calendar(andela_user, event):
+    eventId = event.event_id_in_calendar
```
Comment (Alameen688) [inline server/graphql_schemas/utils/helpers.py]: Kindly use snake_case for consistency. 
```suggestion
    event_id = event.event_id_in_calendar
```

File: server/api/migrations/0023_event_event_id_in_calendar.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    dependencies = [
+        ('api', '0022_event_slack_channel'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='event',
+            name='event_id_in_calendar',
+            field=models.CharField(default=django.utils.timezone.now, max_length=150),
```
Comment (Alameen688) [inline server/api/migrations/0023_event_event_id_in_calendar.py:19]: The default value for `event_id_in_calendar` should not be a timezone value. You can set blank to true in the model instead

File: server/graphql_schemas/event/schema.py [hunk 6]
```diff
@@ -117,7 +117,7 @@ def mutate_and_get_payload(cls, root, info, **input):
             raise GraphQLError("An Error occurred. \n{}".format(e))
 
         slack_token = False
-        if user_profile.slack_tokena:
+        if user_profile.slack_token:
```
Comment (Alameen688) [inline server/graphql_schemas/event/schema.py:115]: Thanks for fixing this 👍

Comment (andrewinsoul) [reply server/graphql_schemas/event/schema.py:115]: You are welcome

File: server/graphql_schemas/utils/helpers.py (hunk 4 above)
Comment (andrewinsoul) [reply server/graphql_schemas/utils/helpers.py]: feedback implemented

File: server/graphql_schemas/utils/helpers.py [hunk 7]
```diff
@@ -213,5 +213,24 @@ def add_event_to_calendar(andela_user, event):
...
+
+def update_event_status_on_calendar(andela_user, event):
+    event_id = event.event_id_in_calendar
+    attendee = andela_user.user.email
+    calendar = build('calendar', 'v3', credentials=andela_user.credential)
+    event_in_calendar = calendar.events().get(calendarId='primary', eventId=event_id).execute()
+    attendees = event_in_calendar['attendees']
+    for attendee in attendees:
+        if attendee['email'] == andela_user.user.email:
+            attendee['responseStatus'] = 'accepted'
```
Comment (Alameen688) [inline server/graphql_schemas/utils/helpers.py]: It seems this only works for users who have already received invites for the event, due to the fact that they were sent invites when an event was created in their category of interest. 
If a user tries to attend an event which they were not sent invites for, due to the fact that it wasn't created under their category of interest, this implementation won't work. 
Kindly help address this. Thanks

**Review by Alameen688 — COMMENTED** [review]
@andrewinsoul Kindly help look into the implementation again. Thanks

File: server/graphql_schemas/utils/helpers.py (hunk 7 above)
Comment (andrewinsoul) [reply server/graphql_schemas/utils/helpers.py]: I have fixed this

**Pull request merged**